cmd/thanos/compact.go

```diff
@@ -220,26 +216,8 @@ func runCompact(
 			cancel()
 		})
 	}
-	// Start metric and profiling endpoints.
-	{
-		router := route.New()
-		ui.New(logger, nil).Register(router)
-
-		mux := http.NewServeMux()
-		registerMetrics(mux, reg)
-		registerProfile(mux)
-		mux.Handle("/", router)
-
-		l, err := net.Listen("tcp", httpAddr)
-		if err != nil {
-			return errors.Wrapf(err, "listen on address %s", httpAddr)
-		}
-
-		g.Add(func() error {
-			return errors.Wrap(http.Serve(l, mux), "serve query")
-		}, func(error) {
-			l.Close()
-		})
+	if err := metricHTTPListenGroup(g, logger, reg, httpBindAddr); err != nil {
```

Comment:
could this not be called / setup in `main.go` for all cmds? seems like most if not all would want this.

Reply:
Yea, but some commands does not need it (CLI only) and some needs some other routes as well (e.g query), so not sure if worth it.